init.go

```diff
@@ -200,3 +209,29 @@ func createDirectories(directories []string) error {
 	}
 	return nil
 }
+
+func shutdownPlugin() {
+	log.Info("Shutting down apidAnalytics plugin")
+
+	// Close all open files and move directories in tmp to staging.
+	bucketMaplock.RLock()
+	defer bucketMaplock.RUnlock()
```

Comment:
deadlock here, don't use `defer`, please check the following comments for details.